### ISSUE TITLE
Dropdown onChange not fired when selectedKey is defined

### ISSUE THREAD

**Issue by dioklecijan** [OfficeDev/office-ui-fabric-react#9310]
### Environment Information 
 
- __Package version(s)__: 6.187.2 (//unpkg.com/office-ui-fabric-react/dist/office-ui-fabric-react.js)
- __Browser and OS versions__: Chrome 74.0.3729.169, Firefox Nightly 68.0a1 

### Please provide a reproduction of the bug in a codepen:
 
Open https://developer.microsoft.com/en-us/fabric/#/controls/web/dropdown and try to change selection for controlled single-select or controlled multi-select example. I can't.
Choose Export to Pen and remove line `selectedKey={this.state.selectedItem}` (`selectedKeys={selectedItems}` for multi-select example) and check can you change selection.

#### Actual behavior:
 
Controlled single-select and multi-select examples at https://developer.microsoft.com/en-us/fabric/#/controls/web/dropdown do not work.

I found that Dropdown does not fire `onChange` event when `selectedKey` or `defaultSelectedKey` is defined.

```js
<Dropdown
        label="Controlled example"
// following two props do not play well together
        selectedKey={selectedItem ? selectedItem.key : undefined}
        onChange={this._onChange}
     ...
/>
```
When I remove `selectedKey` and `defaultSelectedKey`, everything works.

 
#### Expected behavior:
 
`selectedKey(s)` and `defaultSelectedKey(s)` should work together with `onChange`. 

Also, documentation should note `selectedKey` prop in `selectedKeys`description as it does for `defaultSelectedKey`in `defaultSelectedKeys`description. 

Thank you.

**Comment by known-as-bmf** [comment]
related commit: https://github.com/OfficeDev/office-ui-fabric-react/commit/b5c948c84fd5138fca3534e22e6c523edc05e3a4

when a (default)selectedKey(s) is defined, onChange callback is short-circuited.

**Comment by dioklecijan** [comment]
> related commit: [b5c948c](https://github.com/OfficeDev/office-ui-fabric-react/commit/b5c948c84fd5138fca3534e22e6c523edc05e3a4)
> 
> when a (default)selectedKey(s) is defined, onChange callback is short-circuited.

This commit does not solve the problem.

**Comment by known-as-bmf** [comment]
> > related commit: [b5c948c](https://github.com/OfficeDev/office-ui-fabric-react/commit/b5c948c84fd5138fca3534e22e6c523edc05e3a4)
> > when a (default)selectedKey(s) is defined, onChange callback is short-circuited.
> 
> This commit does not solve the problem.

It introduces it.

**Comment by joschect** [comment]
Thanks for reporting this! It was definitely an issue, I fixed it in pr #9315

**Comment by aneeshack4** [comment]
Thanks @joschect for the quick fix!
